exercises/proverb/canonical-data.json

```diff
@@ -0,0 +1,69 @@
+{
+    "cases": [
+        {
+            "cases": [
+                {
+                    "description": "zero pieces",
+                    "expected": "",
+                    "input": [],
+                    "property": "construct"
+                },
+                {
+                    "description": "one piece",
+                    "expected": "And all for the want of a nail.",
+                    "input": [
+                        "nail"
+                    ],
+                    "property": "construct"
+                },
+                {
+                    "description": "two pieces",
+                    "expected": "For want of a nail the shoe was lost.\nAnd all for the want of a nail.",
+                    "input": [
+                        "nail",
+                        "shoe"
+                    ],
+                    "property": "construct"
+                },
+                {
+                    "description": "three pieces",
+                    "expected": "For want of a nail the shoe was lost.\nFor want of a shoe the horse was lost.\nAnd all for the want of a nail.",
+                    "input": [
+                        "nail",
+                        "shoe",
+                        "horse"
+                    ],
+                    "property": "construct"
+                },
+                {
+                    "description": "full proverb",
+                    "expected": "For want of a nail the shoe was lost.\nFor want of a shoe the horse was lost.\nFor want of a horse the rider was lost.\nFor want of a rider the message was lost.\nFor want of a message the battle was lost.\nFor want of a battle the kingdom was lost.\nAnd all for the want of a nail.",
+                    "input": [
```

Comment:
This method of input is not really specified in the description and is an implementation detail.

We either need to: 
* Update the description to specify that you will be provided with a list of items.
* Change the tests so we're just testing for the specified verses. (see also food-chain.)

I think in order to differentiate this exercise from `food-chain` it might be good to use the list of things version.

cc: @ErikSchierboom

Reply:
@Insti I agree that having a different input here makes for a more interesting exercise. The existing exercises are quite similar, so having some variety is nice.

Reply:
So the description should also be updated to clarify the problem.